ext/B/B.xs

```diff
@@ -1422,6 +1422,30 @@ object_2svref(sv)
     PPCODE:
 	ST(0) = sv_2mortal(newRV(sv));
 	XSRETURN(1);
+
+bool
+is_bool(sv)
+    B::SV   sv
+    CODE:
+    RETVAL = SvIsBOOL(sv) ? TRUE : FALSE;
+    OUTPUT:
+        RETVAL
+
+bool
+is_bool_yes(sv)
+    B::SV   sv
+    CODE:
+    RETVAL = SvIsBOOL(sv) && SvTRUE(sv) ? TRUE : FALSE;
```

Comment:
Aside from @leonerd 's comment (which I agree with), this will invoke magic if any on the SV if it started out as a bool.

Other B accessors correctly don't do that, since the intent is to fetch the internal state of the SV.

This (or a replacement TRUE method if the combos are removed) should use SvTRUE_nomg() rather than SvTRUE().

Reply:
yeap I can use SvTRUE_nomg... 
my first implementation was using a PVX check as the `Perl_sv_isbool` does.
But as suggested I changed it to `SvTRUE` and you are right `SvTRUE_nomg` is a better choice
I wonder if the `PVX` check was not the correct way to do it
```
 PERL_STATIC_INLINE bool
 Perl_sv_isbool(pTHX_ const SV *sv)
 {
     return SvIOK(sv) && SvPOK(sv) && SvIsCOW_static(sv) &&
         (SvPVX_const(sv) == PL_Yes || SvPVX_const(sv) == PL_No);
 }
```

Reply:
The `SvPVX_const` test is how the implementation of SvIsBOOL itself is implemented (along with the IsCOW_static check). That function is the one that implements the macro that everyone *else* should be using.

Reply:
> Aside from @leonerd 's comment (which I agree with), this will invoke magic if any on the SV if it started out as a bool.

@tonycoz: Is it possible to have a bool SV with magic on it?